[libc++] Disallow specializing `common_reference` (#141465)
llvm/llvm-project: libcxx/docs/ReleaseNotes/21.rst

- The implementation of ``num_put::do_put`` has been replaced to improve the performance, which can lead to different
  output when printing pointers.

> - User-defined specializations of ``std::common_reference`` are disallowed with the ``[[clang::no_specializations]]``
>   attribute. To customize the common reference type, ``std::basic_common_reference`` should be specialized instead.

Reviewer: ```suggestion
- User-defined specializations of ``std::common_reference`` are diagnosed now. To customize the common reference type, ``std::basic_common_reference`` should be specialized instead.
```
I don't think it's relevant how we do that.
